File: 백준/Gold/1806. 부분합/부분합.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        File file = new File("input.txt");
        BufferedReader br = file.exists()
                ? new BufferedReader(new FileReader(file))
                : new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int S = Integer.parseInt(st.nextToken());

        int[] seq = new int[N];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            seq[i] = Integer.parseInt(st.nextToken());
        }

        int minLen = Integer.MAX_VALUE;
        int sum = 0;
        int left = 0;

        for (int right = 0; right < N; right++) {
            sum += seq[right];

            while (sum >= S) {
                minLen = Math.min(minLen, right - left + 1);
                sum -= seq[left];
                left++;
            }
        }

        System.out.println(minLen == Integer.MAX_VALUE ? 0 : minLen);
    }
}

File: 백준/Gold/1806. 부분합/부분합.js
const fs = require('fs');

// const input = fs.readFileSync('./input.txt').toString().trim().split('\n');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, S] = input[0].split(' ').map(Number);
const seq = input[1].split(' ').map(Number);

let sum = 0;
let left = 0;
let minLength = Infinity;

for (let right = 0; right < N; right++) {
  sum += seq[right];

  while (sum >= S) {
    minLength = Math.min(minLength, right - left + 1);
    sum -= seq[left];
    left++;
  }
}

console.log(minLength === Infinity ? 0 : minLength);
